src/Discovery/DiscoveryDetails/DataDownloadList/ActionButtons/ActionButtons.tsx

```diff
@@ -1,27 +1,30 @@
 import React, { useEffect, useState } from 'react';
+import { RouteComponentProps } from 'react-router-dom';
 import {
   Col, Row, Button, Popover,
 } from 'antd';
-import { useHistory, useLocation } from 'react-router-dom';
 import { DiscoveryConfig } from '../../../DiscoveryConfig';
 import { DiscoveryResource } from '../../../Discovery';
-import { INITIAL_DOWNLOAD_STATUS } from './DownloadUtils/Constants';
 import UseHandleRedirectToLoginClick from './DownloadUtils/UseHandleRedirectToLoginClick';
 import HandleDownloadManifestClick from './DownloadUtils/HandleDownloadManifestClick';
-import DownloadModal from './DownloadModal/DownloadModal';
-import DownloadAllFiles from './DownloadUtils/DownloadAllFiles/DownloadAllFiles';
+import DownloadDataFiles from './DownloadUtils/DownloadDataFiles/DownloadDataFiles';
 import DownloadJsonFile from './DownloadUtils/DownloadJsonFile';
 import DownloadVariableMetadata from './DownloadUtils/DownloadVariableMetadata/DownloadVariableMetadata';
 import './ActionButtons.css';
 import DownloadDataDictionaryInfo from './DownloadUtils/DownloadDataDictionaryInfo';
 import DataDictionaries from '../Interfaces/DataDictionaries';
+import DownloadStatus from '../Interfaces/DownloadStatus';
 
 interface ActionButtonsProps {
   isUserLoggedIn: boolean;
   discoveryConfig: DiscoveryConfig;
   resourceInfo: DiscoveryResource;
-  healLoginNeeded: boolean;
+  healLoginNeeded: string[];
```

Comment:
Suggestion: Rename variable to ```healLoginInfo``` 

In my opinion the name ```healLoginNeeded``` sounds like the name of something that is either true or false

Reply:
same as https://github.com/uc-cdis/data-portal/pull/1492#discussion_r1505321750